### PR TITLE
Use microseconds instead of milliseconds to generate higher throughput

### PR DESCRIPTION
**Pull request by rafaeling**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,7 +166,7 @@ i. e.
   "groups": [
     {
       "group_name": "Frame 1",
-      "cycle_time_ms": 10,
+      "cycle_time_μs": 1000,
```
Comment (erikbosch) [inline README.md:169]: You need to update `in milliseconds` some lines above

File: src/config.rs [hunk 2]
```diff
@@ -21,7 +21,7 @@ pub struct Config {
 #[derive(Deserialize, Clone)]
 pub struct Group {
     pub group_name: String,
-    pub cycle_time_ms: u16,
+    pub cycle_time_μs: u16,
```
Comment (erikbosch) [inline src/config.rs:24]: Is u16 big enough if using microseconds? i.e. at most 65 milliseconds cycle time? (previously 65 seconds)

Comment (rafaeling) [reply src/config.rs:24]: good point, thanks

Comment (wba2hi) [inline src/config.rs:24]: If I understand this correctly u16 similar to uint16 allows a max value of 65_535 (https://doc.rust-lang.org/std/u16/constant.MAX.html) this feels like it's not enough for an interval specified in microseconds?

Comment (rafaeling) [reply src/config.rs]: FIxed

Comment (rafaeling) [reply src/config.rs]: FIxed

**Comment by erikbosch** [comment]
What happens if you use an "old" config file, like the ones in https://github.com/boschglobal/kuksa-perf/tree/fix/cycle_time_microseconds/data

Will you get an error or will it silently just ignore the old setting? I would prefer that it gives an error so that it just not silently ignore the old setting.

**Comment by rafaeling** [comment]
> What happens if you use an "old" config file, like the ones in https://github.com/boschglobal/kuksa-perf/tree/fix/cycle_time_microseconds/data
> 
> Will you get an error or will it silently just ignore the old setting? I would prefer that it gives an error so that it just not silently ignore the old setting.
> 
> Are there more existing scripts that needs to be updated?

Forgot to push the modified files

File: data/data_group_2.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
     },
     {
       "group_name": "Frame 2",
-      "cycle_time_ms": 20,
+      "cycle_time_microseconds": 20,
```
Comment (erikbosch) [inline data/data_group_2.json:14]: Just want to make sure that it is intentional to change semantic meaning of the example, i.e. that we in some cases (like here) keep the numeric value, in other multiply be 100

Comment (rafaeling) [reply data/data_group_2.json]: It was actually non intentional, just left some value, provided more realistic values

**Review by erikbosch — APPROVED** [review]
LGTM

**Review by argerus — CHANGES_REQUESTED** [review]
I think it makes a lot more sense to keep the `"cycle_time_ms"` in order to not break all old configs. In addition, I think it's a lot more common to specify cycle time in milliseconds than microseconds.

TL;DR
Add `"cycle_time_us"` in addition to `"cycle_time_ms"` instead of replacing it.

EDIT: Or skip adding `"cycle_time_us"` altogether and instead have the existing config accept a decimal number for the cycle time, e.g. `"cycle_time_ms": 0.1`.

**Review by argerus — CHANGES_REQUESTED** [review]

**Comment by argerus** [comment]
> What happens if you use an "old" config file, like the ones in https://github.com/boschglobal/kuksa-perf/tree/fix/cycle_time_microseconds/data
> 
> Will you get an error or will it silently just ignore the old setting? I would prefer that it gives an error so that it just not silently ignore the old setting.
> 
> Are there more existing scripts that needs to be updated?

If we need to support specifying the cycle time in microseconds, which seems rather non-realistic, I would much prefer to add this _in addition_ to the existing configuration option, i.e. support both `"cycle_time_ms"` and `"cycle_time_microseconds"`. That has the added benefit of not breaking any existing test setups.
